react-carousel-component/src/components/carousel.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from 'react';
+
+const Carousel = ({ images }) => {
+  const [current, setCurrent] = useState(0);
+  const imagesLength = images.length;
+  const dotOrder = [0, 1, 2, 3];
+
+  const goNextImage = () => {
+    setCurrent(current === imagesLength - 1 ? 0 : current + 1);
+  };
+
+  useEffect(() => {
+    const id = setTimeout(goNextImage, 3000);
+    return () => clearTimeout(id);
+  }, [current]);
+
+  const moveRight = () => {
+    setCurrent(current === imagesLength - 1 ? 0 : current + 1);
+  };
+
+  const moveLeft = () => {
+    setCurrent(current === 0 ? 3 : current - 1);
+  };
```

Comment:
The carousel should not assume how many images it is displaying, you'll need to do your bounds checking based on the length of the images array passed in.